arm/Microsoft.Automation/automationAccounts/jobSchedules/deploy.bicep

```diff
@@ -1,5 +1,5 @@
 @description('Optional. Name of the Automation Account job schedule. Must be a GUID. If not provided, a new GUID is generated.')
-param name string = newGuid()
+param name string = guid(resourceGroup().id, automationAccountName, runbookName, scheduleName)
```

Comment:
That's a tough topic. Tried this but unfortunately doesn't work. Ref: https://docs.microsoft.com/en-us/azure/automation/shared-resources/schedules#create-a-schedule-with-a-resource-manager-template
![image](https://user-images.githubusercontent.com/56914614/143611266-572d9d93-b52d-41bd-ae54-ab93f1a5fa29.png)

You'll get an error if you retry the deployment after deletion. I agree the new guid is not 100% good either as it mines idempotency. The only solution would be to be able to check if the jobschedule with runbook+schedule exists already, if not deploy it, otherwise update it using the existing guid. Not easily achievable with bicep

Reply:
I got error using a new guid for the same combo. Ill revert it.

Reply:
reverted